File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
#include<bits/stdc++.h>
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> map;
        int left = 0, cnt = 0;
        for(int right = 0; right < s.length(); right++) {
            map[s[right]]++;
            while(map[s[right]] > 1) {
                map[s[left]]--;
                left++;
            }
            cnt = max(cnt, right - left + 1);
        }
        return cnt;
    }

};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        l = 0
        map = {} 
        lenm = 0 
        for r in range(len(s)):
            if(s[r] not in map or map[s[r]]<l):
                map[s[r]] = r
                lenm = max(lenm,r-l+1)
            else:
                l = map[s[r]]+1
            map[s[r]] = r
        return lenm

        